app/code/Magento/Developer/Console/Command/patch_template.php.dist

```diff
@@ -15,7 +15,7 @@ use Magento\Framework\Setup\ModuleDataSetupInterface;
 * Patch is mechanism, that allows to do atomic upgrade data changes
 */
 class %class% implements
-    %patchInterface%
+    %patchInterface%%patchRevertableInterface%
```

Comment:
I guess the %patchRevertableInterface% should be on a new line.
Here's an example of what I mean:
```
/**
 * Wrapper for GraphQl NonNull
 */
class NonNull extends \GraphQL\Type\Definition\NonNull implements
    WrappedTypeInterface,
    InputTypeInterface,
    OutputTypeInterface
{
```

Reply:
> I guess the %patchRevertableInterface% should be on a new line.

Not really, @dmytro-ch  :)

`%patchRevertableInterface%` is the alias of `, PatchRevertableInterface`.

So, with the `--revertable=true` flag a result will be:

```php
class TestPatch implements
    DataPatchInterface, PatchRevertableInterface
{
```

Without it:

```php
class TestPatch implements
    DataPatchInterface
{
```

So, I would like to leave this part as it is.

Thank you!

Reply:
I mean that the  `PatchRevertableInterface` should be added on a new line.
Please, check the example above, it shows how the interfaces should be arranged.

According to PSR-2 coding style guide:
{quote}
Lists of implements MAY be split across multiple lines, where each subsequent line is indented once. When doing so, the first item in the list MUST be on the next line, and there MUST be only one interface per line.
{quote}

Reply:
> I mean that the PatchRevertableInterface should be added on a new line.

**done**

**Command 1:**

```
bin/magento setup:db-declaration:generate-patch --revertable=true Magento_Cms TestPatch
```

**Bash output 1:**

```
Patch app/code/Magento/Cms/Setup/Patch/DataTestPatch.php has been successfully generated.
```

**Result 1:**

```
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Magento\Cms\Setup\Patch\Data;

use Magento\Framework\Setup\ModuleDataSetupInterface;
use Magento\Framework\Setup\Patch\DataPatchInterface;
use Magento\Framework\Setup\Patch\PatchRevertableInterface;

/**
* Patch is mechanism, that allows to do atomic upgrade data changes
*/
class TestPatch implements
    DataPatchInterface,
    PatchRevertableInterface
{
    /**
     * @var ModuleDataSetupInterface $moduleDataSetup
     */
    private $moduleDataSetup;

    /**
     * @param ModuleDataSetupInterface $moduleDataSetup
     */
    public function __construct(ModuleDataSetupInterface $moduleDataSetup)
    {
        $this->moduleDataSetup = $moduleDataSetup;
    }

    /**
     * Do Upgrade
     *
     * @return void
     */
    public function apply()
    {
    }

    /**
     * @inheritdoc
     */
    public function revert()
    {
    }

    /**
     * {@inheritdoc}
     */
    public function getAliases()
    {
        return [];
    }

    /**
     * {@inheritdoc}
     */
    public static function getDependencies()
    {
        return [

        ];
    }
}
```

**Command 2:**

```
bin/magento setup:db-declaration:generate-patch --revertable=false Magento_Cms TestPatch
```

**Bash output 2:**

```
Patch app/code/Magento/Cms/Setup/Patch/DataTestPatch.php has been successfully generated.
```

**Result 2:**

```
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Magento\Cms\Setup\Patch\Data;

use Magento\Framework\Setup\ModuleDataSetupInterface;
use Magento\Framework\Setup\Patch\DataPatchInterface;

/**
* Patch is mechanism, that allows to do atomic upgrade data changes
*/
class TestPatch implements DataPatchInterface
{
    /**
     * @var ModuleDataSetupInterface $moduleDataSetup
     */
    private $moduleDataSetup;

    /**
     * @param ModuleDataSetupInterface $moduleDataSetup
     */
    public function __construct(ModuleDataSetupInterface $moduleDataSetup)
    {
        $this->moduleDataSetup = $moduleDataSetup;
    }

    /**
     * Do Upgrade
     *
     * @return void
     */
    public function apply()
    {
    }

    /**
     * {@inheritdoc}
     */
    public function getAliases()
    {
        return [];
    }

    /**
     * {@inheritdoc}
     */
    public static function getDependencies()
    {
        return [

        ];
    }
}
```

**Command 3:**

```
bin/magento setup:db-declaration:generate-patch --revertable=true --type=schema Magento_Cms TestPatch
```

**Bash output 3:**

```
Patch app/code/Magento/Cms/Setup/Patch/SchemaTestPatch.php has been successfully generated.
```

**Result 3:**

```
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Magento\Cms\Setup\Patch\Schema;

use Magento\Framework\Setup\ModuleDataSetupInterface;
use Magento\Framework\Setup\Patch\PatchRevertableInterface;
use Magento\Framework\Setup\Patch\SchemaPatchInterface;

/**
* Patch is mechanism, that allows to do atomic upgrade data changes
*/
class TestPatch implements
    SchemaPatchInterface,
    PatchRevertableInterface
{
    /**
     * @var ModuleDataSetupInterface $moduleDataSetup
     */
    private $moduleDataSetup;

    /**
     * @param ModuleDataSetupInterface $moduleDataSetup
     */
    public function __construct(ModuleDataSetupInterface $moduleDataSetup)
    {
        $this->moduleDataSetup = $moduleDataSetup;
    }

    /**
     * Do Upgrade
     *
     * @return void
     */
    public function apply()
    {
    }

    /**
     * @inheritdoc
     */
    public function revert()
    {
    }

    /**
     * {@inheritdoc}
     */
    public function getAliases()
    {
        return [];
    }

    /**
     * {@inheritdoc}
     */
    public static function getDependencies()
    {
        return [

        ];
    }
}
```

**Command 4:**

```
bin/magento setup:db-declaration:generate-patch --revertable=false --type=schema Magento_Cms TestPatch
```

**Bash output 4:**

```
Patch app/code/Magento/Cms/Setup/Patch/SchemaTestPatch.php has been successfully generated.
```

**Result 4:**

```
<?php
/**
 * Copyright © Magento, Inc. All rights reserved.
 * See COPYING.txt for license details.
 */

namespace Magento\Cms\Setup\Patch\Schema;

use Magento\Framework\Setup\ModuleDataSetupInterface;
use Magento\Framework\Setup\Patch\SchemaPatchInterface;

/**
* Patch is mechanism, that allows to do atomic upgrade data changes
*/
class TestPatch implements SchemaPatchInterface
{
    /**
     * @var ModuleDataSetupInterface $moduleDataSetup
     */
    private $moduleDataSetup;

    /**
     * @param ModuleDataSetupInterface $moduleDataSetup
     */
    public function __construct(ModuleDataSetupInterface $moduleDataSetup)
    {
        $this->moduleDataSetup = $moduleDataSetup;
    }

    /**
     * Do Upgrade
     *
     * @return void
     */
    public function apply()
    {
    }

    /**
     * {@inheritdoc}
     */
    public function getAliases()
    {
        return [];
    }

    /**
     * {@inheritdoc}
     */
    public static function getDependencies()
    {
        return [

        ];
    }
}

```